bdd-generator/src/main/java/com/epam/test_generator/services/AdminService.java

```diff
@@ -21,21 +21,28 @@
     private RoleService roleService;
 
     /**
-     * Changes user role to role specified in userRoleUpdateDTO.
-     * Searching user occurs by e-mail address.
-     * @param userRoleUpdateDTO
+     * Changes user role to role specified in userRoleUpdateDTO. Searching user occurs by e-mail
+     * address.
      */
     public void changeUserRole(UserRoleUpdateDTO userRoleUpdateDTO) {
 
         User userByEmail = userService.getUserByEmail(userRoleUpdateDTO.getEmail());
+        String roleForChange = userRoleUpdateDTO.getRole();
 
         if (userByEmail == null) {
             throw new UnauthorizedException(
-                    "User with email: " + userRoleUpdateDTO.getEmail() + " not found.");
+                "User with email: " + userRoleUpdateDTO.getEmail() + " not found.");
         }
-        Role aNewRole = getRole(userRoleUpdateDTO.getRole());
+        if (userByEmail.getRole().getName().equals("ADMIN")) {
+            throw new BadRoleException("Prohibited to change admin's role");
+        }
+        if (roleForChange.equals("ADMIN")) {
```

Comment:
There are more then one appear of the string "ADMIN". You may make a constant for this role name

Reply:
Final field added